sync.py

```diff
@@ -52,11 +80,27 @@ def all_messages(credentials, full_sync=False) -> int:
 
     service = build("gmail", "v1", credentials=credentials)
 
+    # --- Detect and mark deleted messages --- 
+    print("Checking for deleted messages...")
+    local_active_ids = db.get_active_message_ids()
+    gmail_all_ids = _get_all_gmail_message_ids(service)
+
+    ids_to_mark_deleted = list(local_active_ids - gmail_all_ids)
+    if ids_to_mark_deleted:
+        db.mark_messages_as_deleted(ids_to_mark_deleted)
+        print(f"Marked {len(ids_to_mark_deleted)} messages as deleted.")
+    else:
+        print("No messages found to mark as deleted.")
+    # --- End of deletion check ---
```

Comment:
Would it make sense to batch the messages in the database, taking, for example, 100, and then perform a `messages.list` call on the Gmail service with a query that has the upper and lower bounds set to the timestamps of the first and last message? After that, we can compare the two ID sets and mark the missing IDs from the Gmail service as deleted in the database. This approach would reduce resource consumption when dealing with a large number of files.